lxd/device/disk.go

```diff
@@ -2597,13 +2597,25 @@ func (d *disk) generateVMConfigDrive() (string, error) {
 		}
 	}
 
-	// Append any custom meta-data to our predefined meta-data config.
-	metaData := fmt.Sprintf(`instance-id: %s
-local-hostname: %s
-%s
-`, d.inst.Name(), d.inst.Name(), instanceConfig["user.meta-data"])
+	var metaDataBuilder strings.Builder
 
-	err = os.WriteFile(filepath.Join(scratchDir, "meta-data"), []byte(metaData), 0400)
+	// Append strings to the builder
+	metaDataBuilder.WriteString("instance-id: " + d.inst.Name() + "\n")
+	metaDataBuilder.WriteString("local-hostname: " + d.inst.Name() + "\n")
+
+	// Add keys that are exposed to cloud-init to meta-data so one can use in jinja templates.
+	for key, value := range instanceConfig {
```

Comment:
@hamistao we still need to exclude the `user.meta-data`, `user.user-data`, `user.vendor-data`, `cloud-init.user-data` and `cloud-init.vendor-data` keys here please.

If we are going to need this logic in multiple places suggest having a constant slice of excluded keys.